platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/chassis.py

```diff
@@ -0,0 +1,82 @@
+
+#############################################################################
+# DELLEMC
+#
+# Module contains an implementation of SONiC Platform Base API and
+# provides the platform information
+#
+#############################################################################
+
+try:
+    import os
+    from sonic_platform_base.chassis_base import ChassisBase
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+
+class Chassis(ChassisBase):
+    """
+    DELLEMC Platform-specific Chassis class
+    """
+
+    HWMON_DIR = "/sys/devices/platform/SMF.512/hwmon/"
+    HWMON_NODE = os.listdir(HWMON_DIR)[0]
+    MAILBOX_DIR = HWMON_DIR + HWMON_NODE
+
+    reset_reason_dict = {}
+    reset_reason_dict[11] = [(ChassisBase.REBOOT_CAUSE_POWER_LOSS)]
+    reset_reason_dict[33] = [(ChassisBase.REBOOT_CAUSE_WATCHDOG)]
+    reset_reason_dict[44] = [(ChassisBase.REBOOT_CAUSE_NON_HARDWARE)]
+    reset_reason_dict[55] = [(ChassisBase.REBOOT_CAUSE_NON_HARDWARE)]
+
+    power_reason_dict = {}
+    power_reason_dict[11] = [(ChassisBase.REBOOT_CAUSE_POWER_LOSS)]
+    power_reason_dict[22] = [(ChassisBase.REBOOT_CAUSE_THERMAL_OVERLOAD_CPU)]
+    power_reason_dict[33] = [(ChassisBase.REBOOT_CAUSE_THERMAL_OVERLOAD_ASIC)]
+    power_reason_dict[44] = [(ChassisBase.REBOOT_CAUSE_INSUFFICIENT_FAN)]
+
+    def __init__(self):
+        ChassisBase.__init__(self)
+
+    def get_pmc_register(self, reg_name):
+        rv = 'ERR'
+        mb_reg_file = self.MAILBOX_DIR+'/'+reg_name
+
+        if (not os.path.isfile(mb_reg_file)):
+            print mb_reg_file,  'not found !'
+            return rv
+
+        try:
+            with open(mb_reg_file, 'r') as fd:
+                rv = fd.read()
+        except Exception as error:
+            logging.error("Unable to open ", mb_reg_file, "file !")
+
+        rv = rv.rstrip('\r\n')
+        rv = rv.lstrip(" ")
+        return rv
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the cause of the previous reboot
+        """
+        reset_reason = int(self.get_pmc_register('smf_reset_reason'))
+        power_reason = int(self.get_pmc_register('smf_poweron_reason'))
+
+        # Reset_Reason = 11 ==> PowerLoss
+        # So return the reboot reason from Last Power_Reason Dictionary
+        # If Reset_Reason is not 11 return from Reset_Reason dictionary
+        # Also check if power_reason, reset_reason are valid values by
+        # checking key presence in dictionary else return
+        # REBOOT_CAUSE_HARDWARE_OTHER as the Power_Reason and Reset_Reason
+        # registers returned invalid data
+        if (reset_reason == 11):
+            if (power_reason in self.power_reason_dict):
+                return self.power_reason_dict[power_reason]
+        else:
+            if (reset_reason in self.reset_reason_dict):
+                return self.reset_reason_dict[reset_reason]
+
+        return [(ChassisBase.REBOOT_CAUSE_HARDWARE_OTHER, "Invalid Reason")]
```

Comment:
I understand you're using a tuple in an attempt to provide a reason with REBOOT_CAUSE_HARDWARE_OTHER. however, I'm not a fan of returning a tuple here in this special case and a string for every other case. I think this function needs to consistently return the same data type.

Would it be better if we added a new constant for `REBOOT_CAUSE_HARDWARE_INVALID_REASON`?

Reply:
Hi @jleveque 
Agreed REBOOT_CAUSE_HARDWARE_INVALID_REASON is redundant. 
Removed the special case string. We will fallback to REBOOT_CAUSE_HARDWARE_OTHER in a negative case scenario. Ideally it will never hit that statement,

Thanks
Sridhar.R

Reply:
Can you also please change the return type to just the string, not in a list? The same applies to the values in `reset_reason_dict` and `power_reason_dict`; the function should only ever return one of the predefined strings.

Reply:
Hi @jleveque 
Reverted back to returning as tuple, 
In negative case we return hardware_other with invalid reason

Test Example 

root@sonic:~# show reboot-cause
REBOOT REASON : [('hardware_other', 'Invalid Reason')]               <<<<<<<<<<<<<<<<<<<<
User issued 'reboot' command [User: admin, Time: Wed Jun 26 02:47:21 UTC 2019]